core/lib/dal/src/consensus_dal.rs

```diff
@@ -466,38 +466,67 @@ impl ConsensusDal<'_, '_> {
         )))
     }
 
-    /// Next batch that the attesters should vote for.
+    /// Number of L1 batch that the L2 block belongs to.
+    async fn batch_of_block(
+        &mut self,
+        block: validator::BlockNumber,
+    ) -> anyhow::Result<Option<attester::BatchNumber>> {
+        let Some(row) = sqlx::query!(
+            r#"
+            SELECT
+                l1_batch_number
+            FROM
+                miniblocks
+            WHERE
+                number = $1
+            "#,
+            i64::try_from(block.0).context("overflow")?,
+        )
+        .instrument("batch_of_block")
+        .report_latency()
+        .fetch_optional(self.storage)
+        .await?
+        else {
+            return Ok(None);
+        };
+        let Some(batch) = row.l1_batch_number else {
+            return Ok(None);
```

Comment:
> Although I wonder why it seems to go from 1, when I thought batches start from 0

It is possible because batch 0 is hardcoded. I agree that's an unnecessary assumption for this query.

> I see it's used to query the batch number of the genesis block, and since the status returns the next to vote for regardless of whether it even exists, it would make sense to return the next available batch number without it having to be sealed first.

ok